community: added new document loaders based on dedoc library

alexander1999-hub opened the pull request:
### Description
This pull request added new document loaders to load documents of various formats using [Dedoc](https://github.com/ispras/dedoc):
  - `DedocFileLoader` (determine file types automatically and parse)
  - `DedocPDFLoader` (for `PDF` and images parsing)
  - `DedocAPIFileLoader` (determine file types automatically and parse using Dedoc API without library installation)

[Dedoc](https://dedoc.readthedocs.io) is an open-source library/service that extracts texts, tables, attached files and document structure (e.g., titles, list items, etc.) from files of various formats. The library is actively developed and maintained by a group of developers.

`Dedoc` supports `DOCX`, `XLSX`, `PPTX`, `EML`, `HTML`, `PDF`, images and more.
Full list of supported formats can be found [here](https://dedoc.readthedocs.io/en/latest/#id1).
For `PDF` documents, `Dedoc` allows to determine textual layer correctness and split the document into paragraphs.


### Issue
This pull request extends variety of document loaders supported by `langchain_community` allowing users to choose the most suitable option for raw documents parsing.

### Dependencies
The PR added a new (optional) dependency `dedoc>=2.2.5` ([library documentation](https://dedoc.readthedocs.io)) to the `extended_testing_deps.txt`

### Twitter handle
None

### Add tests and docs
  1. Test for the integration: `libs/community/tests/integration_tests/document_loaders/test_dedoc.py`
  2. Example notebook: `docs/docs/integrations/document_loaders/dedoc.ipynb`
  3. Information about the library: `docs/docs/integrations/providers/dedoc.mdx`

### Lint and test

Done locally:

  - `make format`
  - `make lint`
  - `make integration_tests`
  - `make docs_build` (from the project root)

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+
+EXAMPLE_DOCS_DIRECTORY = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', 'examples')
+FILE_NAMES = [
+    "docusaurus-sitemap.xml",
```
можно оставить только уникальные файлы

alexander1999-hub replied:
Сделал

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+    docs = loader.load()
+
+    assert len(docs) == 18
+    assert docs[0].metadata['file_name'] == 'layout-parser-paper.pdf'
```
проверить lint тесты

alexander1999-hub replied:
Проверил

Согласно [этой](https://python.langchain.com/v0.1/docs/contributing/code/#local-development-dependencies) ссылке, надо установить зависимости командой `poetry install --with lint`. Сами тесты запускаются командой `poetry run ruff .`

alexander1999-hub replied:
Такие команды запускаются у них в тестах
![image](https://github.com/alexander1999-hub/langchain/assets/55749660/26bc5d45-7d45-4729-8e04-aa4fba3ece88)

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+    assert len(docs) == 18
+    assert docs[0].metadata['file_name'] == 'layout-parser-paper.pdf'
+    assert docs[0].metadata['file_type'] == 'application/pdf'
+    assert 'This paper introduces LayoutParser, an open-source\nlibrary' in docs[0].page_content
```
лучше взять одну строку без `\n`

alexander1999-hub replied:
Убрал все проверки на строчки с `\n`

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+    assert 'This paper introduces LayoutParser, an open-source\nlibrary' in docs[0].page_content
+    assert 'layout detection [38, 22], table detection [26]' in docs[1].page_content
+    assert 'LayoutParser: A Uniﬁed Toolkit for DL-Based DIA' in docs[2].page_content
+    assert docs[17].metadata['type'] == 'table'
```
лучше брать первую таблицу (и аттачи)

alexander1999-hub replied:
Есть

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+    assert docs[0].metadata['file_name'] == 'example.json'
+    assert docs[0].metadata['file_type'] == 'application/json'
+    assert docs[0].page_content == 'Bye!'
+    assert docs[1].page_content == 'User 2'
```
лучше проверить только первую ноду

alexander1999-hub replied:
Есть

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+    assert len(docs) == 10
+    assert docs[0].metadata['file_name'] == 'whatsapp_chat.txt'
+    assert docs[0].metadata['file_type'] == 'text/plain'
+    assert docs[0].page_content == '[05.05.23, 15:48:11] James: Hi here\n'
```
можно проверить с помощью `in`

alexander1999-hub replied:
Сделал

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+def test_attachment_handling() -> None:
+    file_name = "fake-email-attachment.eml"
+    file_path = os.path.join(EXAMPLE_DOCS_DIRECTORY, file_name)
+    loader = DedocFileLoader(file_path, split="document", with_attachments=True)
```
можно еще добавить `need_content_analysis=True`

alexander1999-hub replied:
Сделал

NastyBoget commented on libs/community/tests/integration_tests/document_loaders/test_dedoc_loaders.py:
```diff
@@ -0,0 +1,143 @@
+
+from libs.community.langchain_community.document_loaders import (
+    DedocFileLoader,
+    DedocAPIFileLoader,  # не знаю, как этот загрузчик тестировать. требовать от ленгчейна сборки дедка в тестах не хочется и при этом делать запросы на хф тоже
```
можно добавить тест одного простенького файла для нашей демки

alexander1999-hub replied:
Сделал

NastyBoget commented on libs/community/tests/unit_tests/document_loaders/test_imports.py:
```diff
@@ -51,6 +51,9 @@
     "CubeSemanticLoader",
     "DataFrameLoader",
     "DatadogLogsLoader",
+    "DedocAPIFileLoader",
```
добавить дедок в зависимости https://github.com/alexander1999-hub/langchain/blob/master/libs/community/extended_testing_deps.txt

alexander1999-hub replied:
![image](https://github.com/alexander1999-hub/langchain/assets/55749660/fcba1f65-4f6c-45f1-80b3-ef3663339f45)
Судя по тому, что происходит у них в тестах во время ПР, нужно добавить зависимости в toml файл. Больше нигде я не могу найти информации об этом. На всякий случай ещё добавлю в `extended_testing_deps.txt`, но явного упоминания этого файла я нигде не нашёл

alexander1999-hub replied:
Добавил, но собрать окружение не смог
![image](https://github.com/alexander1999-hub/langchain/assets/55749660/dfe0a3ec-52c7-4e37-9dac-5e0e1ec3af13)

NastyBoget committed:
community: Add dedoc support

tibor-reiss committed:
core[patch]: Update typing for template format to include jinja2 as a Literal (#24144)

Fixes #23929 via adjusting the typing

Dobiichi-Origami committed:
community[patch]: change the class of `qianfan_ak` and `qianfan_sk` parameters (#24293)

- **Description:** we changed the class of two parameters to fix a bug,
which causes validation failure when using QianfanEmbeddingEndpoint

leo-gan committed:
core[patch]: docstrings `langchain_core/` files update (#24285)

Added missed docstrings. Formatted docstrings to the consistent form.

amcastror committed:
Update retrievers.ipynb (#24289)

Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core,
experimental, etc. is being modified. Use "docs: ..." for purely docs
changes, "templates: ..." for template changes, "infra: ..." for CI
changes.
  - Example: "community: add foobar LLM"


- [ ] **PR message**: ***Delete this entire checklist*** and replace
with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!


- [ ] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

bimslab committed:
baidu qianfan AiMessage with usage_metadata (#24288)

add usage_metadata to qianfan AIMessage. Thanks

shenhai-ran committed:
core[patch]: Add encoding options when create prompt template from a file (#24054)

- Uses default utf-8 encoding for loading prompt templates from file

---------

Co-authored-by: Eugene Yurtsev <eyurtsev@gmail.com>

2020uce0047 committed:
community[patch]: Update TavilySearch to use TavilyClient instead of the deprecated Client (#24270)

On using TavilySearchAPIRetriever with any conversation chain getting
error :

`TypeError: Client.__init__() got an unexpected keyword argument
'api_key'`

It is because the retreiver class is using the depreciated `Client`
class, `TavilyClient` need to be used instead.

---------

Co-authored-by: Eugene Yurtsev <eyurtsev@gmail.com>
Co-authored-by: Eugene Yurtsev <eugene@langchain.dev>

RafaelXokito committed:
Docs: Fix Concepts Integration Tools Link (#24301)

- **Description:** This PR fix concepts integrations tools link.

- **Issue:** Fixes issue #24112

---------

Co-authored-by: ccurme <chester.curme@gmail.com>

leo-gan committed:
core: docstrings `documents` (#23506)

Added missed docstrings. Formatted docstrings to the consistent form.

mazen-r committed:
docs: fix parameter typo in scrapfly loader docs (#24307)

Fixed wrong parameter typo in
[ScrapflyLoader](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/document_loaders/scrapfly.py)
docs, where `ignore_scrape_failures` is used instead of
`continue_on_failure`.

- Description: Fix wrong param typo in ScrapflyLoader docs.

rick-github committed:
community[patch]: Fix docstring for ollama parameter "keep_alive" (#23973)

Fix doc-string for ollama integration

bovlb committed:
community[minor]: Add ApertureDB as a vectorstore (#24088)

Thank you for contributing to LangChain!

- [X] *ApertureDB as vectorstore**: "community: Add ApertureDB as a
vectorestore"

- **Description:** this change provides a new community integration that
uses ApertureData's ApertureDB as a vector store.
    - **Issue:** none
    - **Dependencies:** depends on ApertureDB Python SDK
    - **Twitter handle:** ApertureData

- [X] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.

Integration tests rely on a local run of a public docker image.
Example notebook additionally relies on a local Ollama server.

- [X] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

All lint tests pass.

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

---------

Co-authored-by: Gautam <gautam@aperturedata.io>

Anush008 committed:
qdrant: Fixed typos in Qdrant vectorstore docs (#24312)

## Description 

As that title goes.

baskaryan committed:
core[patch]: fix Typing.cast import (#24313)

Fixes #24287

Commit:
openai: raw response headers (#24150)

leo-gan committed:
core: docstrings `vectorstores` update (#24281)

Added missed docstrings. Formatted docstrings to the consistent form.

---------

Co-authored-by: Erick Friis <erick@langchain.dev>

pbharti0831 committed:
Cookbook for applying RAG locally using open source models and tools on CPU (#24284)

This cookbook guides user to implement RAG locally on CPU using
langchain tools and open source models. It enables Llama2 model to
answer queries about Intel Q1 2024 earning release using RAG pipeline.

Main libraries are langchain, llama-cpp-python and gpt4all.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

---------

Co-authored-by: Sriragavi <sriragavi.r@intel.com>
Co-authored-by: Chester Curme <chester.curme@gmail.com>

Srijan-D committed:
Upgraded shaleprotocol to use langchain v0.2 removed deprecated classes (#24320)

Description: Added support for langchain v0.2 for shale protocol.
Replaced LLMChain with Runnable interface which allows any two Runnables
to be 'chained' together into sequences. Also added
StreamingStdOutCallbackHandler. Callback handler for streaming.
Issue: None
Dependencies: None.

Commit:
docs: readthedocs deprecation fix (#24321)

https://about.readthedocs.com/blog/2024/07/addons-by-default/#how-does-it-affect-my-projects

we use build.command so we're already using addons, so I think this is
it

eyurtsev committed:
Docs: Add how to dispatch custom callback events (#24278)

* Add how-to guide for dispatching custom callback events.
* Add links from index to the how to guide
* Add link from streaming from within a tool
* Update versionadded to correct release
https://github.com/langchain-ai/langchain/releases/tag/langchain-core%3D%3D0.2.15

Commit:
infra: prerelease dep checking on release (#23269)

baskaryan committed:
core[patch]: Release 0.2.20 (#24322)

Commit:
infra: fix release prerelease deps bug (#24323)

RafaelXokito committed:
neo4j: Fix test for order-insensitive comparison and floating-point precision issues (#24338)

**Description:** 
This PR addresses two main issues in the `test_neo4jvector.py`:
1. **Order-insensitive Comparison:** Modified the
`test_retrieval_dictionary` to ensure that it passes regardless of the
order of returned values by parsing `page_content` into a structured
format (dictionary) before comparison.
2. **Floating-point Precision:** Updated
`test_neo4jvector_relevance_score` to handle minor floating-point
precision differences by using the `isclose` function for comparing
relevance scores with a relative tolerance.

Errors addressed:

- **test_neo4jvector_relevance_score:**
  ```
AssertionError: assert [(Document(page_content='foo', metadata={'page':
'0'}), 1.0000014305114746), (Document(page_content='bar',
metadata={'page': '1'}), 0.9998371005058289),
(Document(page_content='baz', metadata={'page': '2'}),
0.9993508458137512)] == [(Document(page_content='foo', metadata={'page':
'0'}), 1.0), (Document(page_content='bar', metadata={'page': '1'}),
0.9998376369476318), (Document(page_content='baz', metadata={'page':
'2'}), 0.9993523359298706)]
At index 0 diff: (Document(page_content='foo', metadata={'page': '0'}),
1.0000014305114746) != (Document(page_content='foo', metadata={'page':
'0'}), 1.0)
  Full diff:
  - [(Document(page_content='foo', metadata={'page': '0'}), 1.0),
+ [(Document(page_content='foo', metadata={'page': '0'}),
1.0000014305114746),
? +++++++++++++++
- (Document(page_content='bar', metadata={'page': '1'}),
0.9998376369476318),
? ^^^ ------
+ (Document(page_content='bar', metadata={'page': '1'}),
0.9998371005058289),
? ^^^^^^^^^
- (Document(page_content='baz', metadata={'page': '2'}),
0.9993523359298706),
? ----------
+ (Document(page_content='baz', metadata={'page': '2'}),
0.9993508458137512),
? ++++++++++
  ]
  ```

- **test_retrieval_dictionary:**
  ```
AssertionError: assert [Document(page_content='skills:\n- Python\n- Data
Analysis\n- Machine Learning\nname: John\nage: 30\n')] ==
[Document(page_content='skills:\n- Python\n- Data Analysis\n- Machine
Learning\nage: 30\nname: John\n')]
At index 0 diff: Document(page_content='skills:\n- Python\n- Data
Analysis\n- Machine Learning\nname: John\nage: 30\n') !=
Document(page_content='skills:\n- Python\n- Data Analysis\n- Machine
Learning\nage: 30\nname: John\n')
  Full diff:
- [Document(page_content='skills:\n- Python\n- Data Analysis\n- Machine
Learning\nage: 30\nname: John\n')]
? ---------
+ [Document(page_content='skills:\n- Python\n- Data Analysis\n- Machine
Learning\nage: John\nage: 30\n')]
? +++++++++
  ```

gnumoksha committed:
docs(infobip.ipynb): fix typo (#24328)

Srijan-D committed:
removed deprecated classes from pipelineai.ipynb, added support for LangChain v0.2 for PipelineAI integration (#24333)

Description: added support for LangChain v0.2 for PipelineAI
integration. Removed deprecated classes and incorporated support for
LangChain v0.2 to integrate with PipelineAI. Removed LLMChain and
replaced it with Runnable interface. Also added StrOutputParser, that
parses LLMResult into the top likely string.

Issue: None

Dependencies: None.

---------

Co-authored-by: Chester Curme <chester.curme@gmail.com>

maang-h committed:
docs: Add ChatBaichuan docstrings (#24348)

- **Description:** Add ChatBaichuan rich docstrings.
- **Issue:** the issue #22296

RafaelXokito committed:
Neo4j: Update with non-deprecated cypher methods, and new method to associate relationship embeddings (#23725)

**Description:** At the moment neo4j wrapper is using setVectorProperty,
which is deprecated
([link](https://neo4j.com/docs/operations-manual/5/reference/procedures/#procedure_db_create_setVectorProperty)).
I replaced with the non-deprecated version.

Neo4j recently introduced a new cypher method to associate embeddings
into relations using "setRelationshipVectorProperty" method. In this PR
I also implemented a new method to perform this association maintaining
the same format used in the "add_embeddings" method which is used to
associate embeddings into Nodes.
I also included a test case for this new method.

baskaryan committed:
langchain[patch]: Release 0.2.9 (#24327)

lmoros-DB committed:
community: Fix SQLDatabse.from_databricks issue when ran from Job (#24346)

- Description: When SQLDatabase.from_databricks is ran from a Databricks
Workflow job, line 205 (default_host = context.browserHostName) throws
an ``AttributeError`` as the ``context`` object has no
``browserHostName`` attribute. The fix handles the exception and sets
the ``default_host`` variable to null

---------

Co-authored-by: lmorosdb <lmorosdb>
Co-authored-by: Eugene Yurtsev <eugene@langchain.dev>

ccurme committed:
openai[patch]: fix key collision and _astream (#24345)

Fixes small issues introduced in
https://github.com/langchain-ai/langchain/pull/24150 (unreleased).

baskaryan committed:
community[patch]: Release 0.2.8 (#24354)

Commit:
pinecone: release 0.1.2 (#24355)

Commit:
openai: release 0.1.17 (#24361)

eyurtsev committed:
core[patch]: Fix one unit test for chat prompt template (#24362)

Minor change that fixes a unit test that had missing assertions.

Commit:
milvus: release 0.1.2 (#24365)

caseyclements committed:
partners/mongodb : Significant MongoDBVectorSearch ID enhancements (#23535)

## Description

This pull-request improves the treatment of document IDs in
`MongoDBAtlasVectorSearch`.

Class method signatures of add_documents, add_texts, delete, and
from_texts
now include an `ids:Optional[List[str]]` keyword argument permitting the
user
greater control. 
Note that, as before, IDs may also be inferred from
`Document.metadata['_id']`
if present, but this is no longer required,
IDs can also optionally be returned from searches.

This PR closes the following JIRA issues.

* [PYTHON-4446](https://jira.mongodb.org/browse/PYTHON-4446)
MongoDBVectorSearch delete / add_texts function rework
* [PYTHON-4435](https://jira.mongodb.org/browse/PYTHON-4435) Add support
for "Indexing"
* [PYTHON-4534](https://jira.mongodb.org/browse/PYTHON-4534) Ensure
datetimes are json-serializable

---------

Co-authored-by: Erick Friis <erick@langchain.dev>

hmasdev committed:
langchain[patch]: fix wrong `dict` key in `OutputFixingParser`, `RetryOutputParser` and `RetryWithErrorOutputParser` (#23967)

# Description
This PR aims to solve a bug in `OutputFixingParser`, `RetryOutputParser`
and `RetryWithErrorOutputParser`
The bug is that the wrong keyword argument was given to `retry_chain`.
The correct keyword argument is 'completion', but 'input' is used.

This pull request makes the following changes:
1. correct a `dict` key given to `retry_chain`;
2. add a test when using the default prompt.
   - `NAIVE_FIX_PROMPT` for `OutputFixingParser`;
   - `NAIVE_RETRY_PROMPT` for `RetryOutputParser`;
   - `NAIVE_RETRY_WITH_ERROR_PROMPT` for `RetryWithErrorOutputParser`;
3. ~~add comments on `retry_chain` input and output types~~ clarify
`InputType` and `OutputType` of `retry_chain`

# Issue
The bug is pointed out in
https://github.com/langchain-ai/langchain/pull/19792#issuecomment-2196512928

---------

Co-authored-by: Erick Friis <erick@langchain.dev>

bricefotzo committed:
community: support advanced text extraction options for pdf documents (#20265)

**Description:** 
- Updated constructors in PyPDFParser and PyPDFLoader to handle
`extraction_mode` and additional kwargs, aligning with the capabilities
of `PageObject.extract_text()` from pypdf.

- Added `test_pypdf_loader_with_layout` along with a corresponding
example text file to validate layout extraction from PDFs.

**Issue:** fixes #19735 

**Dependencies:** This change requires updating the pypdf dependency
from version 3.4.0 to at least 4.0.0.

Additional changes include the addition of a new test
test_pypdf_loader_with_layout and an example text file to ensure the
functionality of layout extraction from PDFs aligns with the new
capabilities.

---------

Co-authored-by: Bagatur <22008038+baskaryan@users.noreply.github.com>
Co-authored-by: Bagatur <baskaryan@gmail.com>
Co-authored-by: Erick Friis <erick@langchain.dev>

eyurtsev committed:
core[patch]: Fix regression requiring input_variables in few chat prompt templates (#24360)

* Fix regression that requires users passing input_variables=[].

* Regression introduced by my own changes to this PR:
https://github.com/langchain-ai/langchain/pull/22851

nithishr committed:
couchbase: Add chat message history (#24356)

**Description:** : Add support for chat message history using Couchbase

- [x] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [x] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

---------

Co-authored-by: Nithish Raghunandanan <nithishr@users.noreply.github.com>

baskaryan committed:
openai[patch]: use model_name in AzureOpenAI.ls_model_name (#24366)

Commit:
openai: release 0.1.18 (#24369)

hinthornw committed:
core[patch]: add InjectedToolArg annotation (#24279)

```python
from typing_extensions import Annotated
from langchain_core.tools import tool, InjectedToolArg
from langchain_anthropic import ChatAnthropic

@tool
def multiply(x: int, y: int, not_for_model: Annotated[dict, InjectedToolArg]) -> str:
    """multiply."""
    return x * y 

ChatAnthropic(model='claude-3-sonnet-20240229',).bind_tools([multiply]).invoke('5 times 3').tool_calls
'''
-> [{'name': 'multiply',
  'args': {'x': 5, 'y': 3},
  'id': 'toolu_01Y1QazYWhu4R8vF4hF4z9no',
  'type': 'tool_call'}]
'''
```

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>
Co-authored-by: Bagatur <22008038+baskaryan@users.noreply.github.com>

baskaryan committed:
core[patch]: Release 0.2.21 (#24372)

Commit:
infra: 15s retry wait on test pypi (#24375)

baskaryan committed:
docs: tools concepts  (#24368)

baskaryan committed:
docs: clean up tool how to titles (#24373)

eyurtsev committed:
core[patch]: Update API reference for astream events (#24359)

Update the API reference for astream events to include information about
custom events.

kharenzze committed:
Community[patch]: Missing "stream" parameter in cloudflare_workersai (#23987)

### Description
Missing "stream" parameter. Without it, you'd never receive a stream of
tokens when using stream() or astream()

### Issue
No existing issue available

NastyBoget committed:
review fixes

eyurtsev committed:
experimental[patch]: block a few more things from PALValidator (#24379)

* Please see security warning already in existing class.
* The approach here is fundamentally insecure as it's relying on a block
  approach rather than an approach based on only running allowed nodes.
So users should only use this code if its running from a properly
sandboxed  environment.

Raj725 committed:
community[minor]: [PebbloSafeLoader] Rename loader type and add SharePointLoader to supported loaders (#24393)

Thank you for contributing to LangChain!

- [x] **PR title**: [PebbloSafeLoader] Rename loader type and add
SharePointLoader to supported loaders
    - **Description:** Minor fixes in the PebbloSafeLoader:
        - Renamed the loader type from `remote_db` to `cloud_folder`.
- Added `SharePointLoader` to the list of loaders supported by
PebbloSafeLoader.
    - **Issue:** NA
    - **Dependencies:** NA
- [x] **Add tests and docs**: NA

ZhangShenao committed:
community[patch]: Fix spelling error in ConversationVectorStoreTokenBufferMemory doc-string (#24385)

Fix word spelling error in `ConversationVectorStoreTokenBufferMemory`

joaodinissf committed:
docs: remove redundant `--quiet` option in `pip install` command (#24397)

- **Description:** Removes a redundant option in a `pip install` command
in the documentation.
- **Issue:** N/A
- **Dependencies:** N/A

bimslab committed:
[fix] baidu qianfan AiMessage with usage_metadata (#24389)

make AIMessage usage_metadata has error

caseyclements committed:
docs: Adds MongoDBAtlasVectorSearch to VectorStore list compatible with Indexing API (#24374)

Adds MongoDBAtlasVectorSearch to list of VectorStores compatible with
the Indexing API.

(One line change.)

As of `langchain-mongodb = "0.1.7"`, the requirements that the
VectorStore have both add_documents and delete methods with an ids kwarg
is satisfied. #23535 contains the implementation of that, and has been
merged.

lunara-kim committed:
community: fix tool appending logic and update planner prompt in OpenAPI agent toolkit (#24384)

**Description:**
- Updated the format for the 'Action' section in the planner prompt to
ensure it must be one of the tools without additional words. Adjusted
the phrasing from "should be" to "must be" for clarity and
enforceability.
- Corrected the tool appending logic in the
`_create_api_controller_agent` function to ensure that
`RequestsDeleteToolWithParsing` and `RequestsPatchToolWithParsing` are
properly added to the tools list for "DELETE" and "PATCH" operations.

**Issue:** #24382

**Dependencies:** None

**Twitter handle:** @lunara_x

---------

Co-authored-by: Chester Curme <chester.curme@gmail.com>

NastyBoget committed:
review fixes part 2

hwchase17 committed:
docs: improve docs on tools (#24404)

Co-authored-by: Bagatur <22008038+baskaryan@users.noreply.github.com>
Co-authored-by: Eugene Yurtsev <eyurtsev@gmail.com>

ccurme committed:
docs: update tool calling how-to to pass functions to bind_tools (#24402)

Commit:
mongodb: release 0.1.7 (#24409)

Srijan-D committed:
Adding LangChain v0.2 support for nvidia ai endpoint, langchain-nvidia-ai-endpoints. Removed deprecated classes from nvidia_ai_endpoints.ipynb (#24411)

Description: added support for LangChain v0.2 for nvidia ai endpoint.
Implremented inMemory storage for chains using
RunnableWithMessageHistory which is analogous to using
`ConversationChain` which was used in v0.1 with the default
`ConversationBufferMemory`. This class is deprecated in favor of
`RunnableWithMessageHistory` in LangChain v0.2

Issue: None

Dependencies: None.

---------

Co-authored-by: Chester Curme <chester.curme@gmail.com>

denver1117 committed:
Add a `ListRerank` document compressor (#13311)

- **Description:** This PR adds a new document compressor called
`ListRerank`. It's derived from `BaseDocumentCompressor`. It's a near
exact implementation of introduced by this paper: [Zero-Shot Listwise
Document Reranking with a Large Language
Model](https://arxiv.org/pdf/2305.02156.pdf) which it finds to
outperform pointwise reranking, which is somewhat implemented in
LangChain as
[LLMChainFilter](https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/retrievers/document_compressors/chain_filter.py).
- **Issue:** None
- **Dependencies:** None
- **Tag maintainer:** @hwchase17 @izzymsft
- **Twitter handle:** @HarrisEMitchell

Notes:
1. I didn't add anything to `docs`. I wasn't exactly sure which patterns
to follow as [cohere reranker is under
Retrievers](https://python.langchain.com/docs/integrations/retrievers/cohere-reranker)
with other external document retrieval integrations, but other
contextual compression is
[here](https://python.langchain.com/docs/modules/data_connection/retrievers/contextual_compression/).
Happy to contribute to either with some direction.
2. I followed syntax, docstrings, implementation patterns, etc. as well
as I could looking at nearby modules. One thing I didn't do was put the
default prompt in a separate `.py` file like [Chain
Filter](https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/retrievers/document_compressors/chain_filter_prompt.py)
and [Chain
Extract](https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/retrievers/document_compressors/chain_extract_prompt.py).
Happy to follow that pattern if it would be preferred.

---------

Co-authored-by: Harrison Chase <hw.chase.17@gmail.com>
Co-authored-by: Bagatur <baskaryan@gmail.com>
Co-authored-by: Chester Curme <chester.curme@gmail.com>

prakul committed:
docs: Update mongodb README.md (#24412)

Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core,
experimental, etc. is being modified. Use "docs: ..." for purely docs
changes, "templates: ..." for template changes, "infra: ..." for CI
changes.
  - Example: "community: add foobar LLM"


- [ ] **PR message**: ***Delete this entire checklist*** and replace
with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!


- [ ] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

eyurtsev committed:
core[minor]: Support all versions of pydantic base model in argsschema (#24418)

This adds support to any pydantic base model for tools.

The only potential issue is that `get_input_schema()` will not always
return a v1 base model.

eyurtsev committed:
standard-tests[patch]: Add pytest assert rewrites (#24408)

This will surface nice error messages in subclasses that fail assertions.

nfcampos committed:
core: In ensure_config don't copy dunder configurable keys to metadata (#24420)

hinthornw committed:
[Evaluation] Pass in seed directly (#24403)

adding test rn

onesolpark committed:
Mask API key for ChatOpenAI based chat_models (#14293)

- **Description**: Mask API key for ChatOpenAi based chat_models
(openai, azureopenai, anyscale, everlyai).
Made changes to all chat_models that are based on ChatOpenAI since all
of them assumes that openai_api_key is str rather than SecretStr.
  - **Issue:**: #12165 
  - **Dependencies:**  N/A
  - **Tag maintainer:** @eyurtsev
  - **Twitter handle:** N/A

---------

Co-authored-by: Chester Curme <chester.curme@gmail.com>

NastyBoget committed:
documentation and dependencies fix
